Automattic/jetpack
# VideoPress: Return a highly compatible video file format for non-VideoPress players

Fixes #

<img width="510" alt="Screen Shot 2022-02-03 at 5 18 27 PM" src="https://user-images.githubusercontent.com/4081020/152439663-771049b7-8f82-4105-8d38-2b8bd2ec240a.png">
<img width="1059" alt="Screen Shot 2022-02-03 at 5 27 34 PM" src="https://user-images.githubusercontent.com/4081020/152439751-4baee85d-521a-4124-b77d-59faa456df3b.png">


This PR introduces support for the new "compatibility" video format being introduced to new VideoPress uploads. This video file allows non-VideoPress players to display a video file that will be compatible in most (well all modern browsers) instead of the originally uploaded video file (which could be an `.mov` file which wouldn't play in Firefox for example).

This essentially returns functionality for 3rd party players back to the way it was prior to 10.2 and the introduction of HLS streaming in VideoPress, but with a higher resolution (1080p vs 720p before) so there shouldn't be any more compatibility issues with these players, and the playback should look better for customers uploading higher resolution videos.

#### Changes proposed in this Pull Request:

* If available, the "compat" video format is returned to themes, plugins and blocks that call `wp_get_attachment_url()`
* If "compat" is available, a new "Original File URL" field is displayed in the Media Library details window so the original url can be copied and shared if necessary.

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?


nope

#### Testing instructions:






* ~~Apply D74319-code to your sandbox if it hasn't been deployed yet~~ deployed!
* Upload a video to the Media Library
* Create a post with a `/cover` block and `/story` block and publish it
* The public page's video urls should point to the `COMPAT` video format (ex `<filename>_hd_1080p_compat.mp4`)
* View the attachment in the Media Library
* The new "Original File Url" filed should be visible with the original file and the "File Url" field should have the COMPAT format
* Add a header video via your theme (if supported)
* View the public page, it should be using the `COMPAT` format
